homeassistant/components/verisure/alarm_control_panel.py

```diff
@@ -12,32 +12,27 @@
     SUPPORT_ALARM_ARM_AWAY,
     SUPPORT_ALARM_ARM_HOME,
 )
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
     STATE_ALARM_ARMED_AWAY,
     STATE_ALARM_ARMED_HOME,
     STATE_ALARM_DISARMED,
     STATE_ALARM_PENDING,
 )
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.entity import Entity
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-from .const import CONF_ALARM, CONF_GIID, DOMAIN, LOGGER
+from .const import CONF_GIID, DOMAIN, LOGGER
 from .coordinator import VerisureDataUpdateCoordinator
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: dict[str, Any],
-    add_entities: Callable[[list[Entity], bool], None],
-    discovery_info: dict[str, Any] | None = None,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[list[VerisureAlarm]], None],
```

Comment:
Not for this PR but we should really export the type for `async add entities` from EntityPlatform.

Reply:
Yeah, we should :(
It's getting messy 👍

Reply:
Adjusted the PR a bit, to be more future proof and more correct.